mlflow/server/js/src/experiment-tracking/components/artifact-view-components/ShowArtifactAudioView.js

```diff
@@ -0,0 +1,168 @@
+import React, { useEffect, useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { getSrc } from './ShowArtifactPage';
+import { getArtifactContent } from '../../../common/utils/ArtifactUtils';
+import WaveSurfer from 'wavesurfer.js';
+
+function formatTimecode(seconds) {
+  return new Date(1000 * seconds).toISOString().substring(11, 19);
+}
+
+const ShowArtifactAudioView = ({ runUuid, path, getArtifact }) => {
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState();
+  const [waveformPainting, setWaveformPainting] = useState();
+  const [playing, setPlaying] = useState(false);
+  const [audioDuration, setAudioDuration] = useState(0);
+  const [timeElapsed, setTimeElapsed] = useState(0);
+  const waveformRef = useRef();
+  const [waveform, setWaveform] = useState();
+
+  useEffect(() => {
+    resetState();
+    fetchArtifacts({ path, runUuid, getArtifact });
+    return () => {
+      if (waveform) {
+        waveform.destroy();
+      }
+    };
+  }, [runUuid, path]);
+
+  function resetState() {
+    setPlaying(false);
+    setAudioDuration(0);
+    setTimeElapsed(0);
+    setLoading(true);
+    setError();
+  }
+
+  function loadAudio(artifact) {
+    let newWaveform;
+    try {
+      newWaveform = WaveSurfer.create({
+        waveColor: '#1890ff',
+        progressColor: '#2374BB',
+        cursorColor: '#333333',
+        container: waveformRef.current,
+        responsive: true,
+        height: 592,
```

Comment:
Any advice on how to make this value dynamic? I tried with using references (`ref`) and getting the parent's height with `ref.current.parentElement.offsetHeight()` but I just don't know React well enough to get this to work. It should just be the height of the parent minus the height of the `audioInfo` element which can be a constant of `42px`.